docs/getting_started/running_a_workflow_locally.md

```diff
@@ -0,0 +1,168 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+---
+
+# Running a workflow locally
+
+You can run a workflow locally in two ways:
+
+* **{ref}`In a local Python environment <getting_started_running_workflow_local_python_environment>`:** To develop and test your code quickly without the overhead of setting up a local Flyte cluster, you can run your workflow in a local Python environment.
+* **{ref}`In a local Flyte cluster <getting_started_running_workflow_local_cluster>`:** If your organization has deployed a remote Flyte cluster, or if you want to test a Flyte cluster setup locally before committing to a remote deployment, you can run your workflows in a local demo Flyte cluster.
+
+(getting_started_running_workflow_local_python_environment)=
+
+## Running a workflow in a local Python environment
+
+### Prerequisites
+
+* {doc}`Install development tools <installing_development_tools>`
+* {doc}`Create a Flyte project <creating_a_flyte_project>`
+
+### Steps
+
+1. On the command line, navigate to the workflows directory of your Flyte project:
+```{prompt} bash $
+cd my_project/workflows
+```
+2. Run the workflow with `pyflyte run`:
+```{prompt} bash $
+pyflyte run example.py wf
+```
+
+:::{note}
+While you can run the example file like a Python script with `python hello_world.py`, we recommend using `pyflyte run` instead. To run the file like a Python script, you would have to add a `main` module conditional at the end of the script:
+```python
+if __name__ == "__main__":
+    print(hello_world_wf())
+```
+
+Your code would become even more verbose if you wanted to pass arguments to the workflow:
+```python
+if __name__ == "__main__":
+    from argparse import ArgumentParser
+
+    parser = ArgumentParser()
+    parser.add_argument("--name", type=str)
+
+    args = parser.parse_args()
+    print(hello_world_wf(name=args.name))
+```
+:::
```

Comment:
This shouldn't be needed. `pyflyte run file.py hello_world_wf` should ask you to enter the `--name` already...

Reply:
Understood, the purpose of this note is to explain why we recommend using `pyflyte run` rather than `python example.py` and is pulled from the [existing docs](https://docs.flyte.org/projects/cookbook/en/latest/index.html), but I can remove it if it seems confusing.